hpx/runtime/serialization/raw_ptr.hpp

```diff
@@ -0,0 +1,120 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_RAW_PTR_HPP
+#define HPX_SERIALIZATION_RAW_PTR_HPP
+
+#include <hpx/runtime/serialization/detail/pointer.hpp>
+
+namespace hpx { namespace serialization {
+
+    namespace detail
+    {
+        template <class T>
+        struct raw_ptr_type
+        {
+            typedef T element_type;
+
+            raw_ptr_type(T* t)
+                : t(t)
+            {}
+
+            raw_ptr_type operator=(const raw_ptr_type& rhs)
+            {
+                if (this == &rhs) return *this;
+                t = rhs.t;
+                return *this;
+            }
+
+            T* get() const
+            {
+                return t;
+            }
+
+            T& operator*() const
+            {
+                return *t;
+            }
+
+            operator bool() const
+            {
+                return t != 0;
+            }
+
+        private:
+            T* t;
+        };
+
+        template <class T>
+        struct raw_ptr_proxy
+        {
+            raw_ptr_proxy(T*& t):
+                t(t)
+            {}
+
+            raw_ptr_proxy(T* const & t):
+                t(const_cast<T*&>(t))
```

Comment:
K-ballo: I used this constructor overload to allow user to pass const pointer to something when saving:
```A* const p = new A;
ar << raw_ptr(p);```
There are several const_casts in serialization which do the same thing. Basically they come out because of nonconst interface of templated serialize method.